src/hotspot/share/gc/shared/barrierSetNMethod.cpp

```diff
@@ -67,10 +61,24 @@ bool BarrierSetNMethod::supports_entry_barrier(nmethod* nm) {
 }
 
 bool BarrierSetNMethod::nmethod_entry_barrier(nmethod* nm) {
+  class OopKeepAliveClosure : public OopClosure {
+  public:
+    virtual void do_oop(oop* p) {
+      // Loads on nmethod oops are phantom strength. The intend of the load
+      // is to just read the oop, and then explicitly keep it alive w.r.t.
+      // concurrent marking. Using the keep alive side effects of a normal
+      // phantom load is less explicit, and doesn't actually do anything
+      // unless the returned value is used as an oop.
```

Comment:
I think this is bit vague. Would it be OK if we explicitly spelled out the problem with the LoadOopProxy? Maybe something like this (feel free to tweak the actual text):
```
// Loads on nmethod oops are phantom strength.
//
// Note: that we could have used NativeAccess<ON_PHANTOM_OOP_REF>::oop_load(p),
// but that would have *required* us to convert the returned LoadOopProxy to an oop,
// or else keep alive load barrier will never be called. It's the LoadOopProxy-to-oop
// conversion that performs the load barriers. This is too subtle, so we instead
// perform an explicit keep alive call.
```

Reply:
I used your comment but removed the colon in "Note: that". Thanks for the review!